doc/DeveloperGuide.md

```diff
@@ -87,6 +87,42 @@ Use case ends.
 
 > 3a1. AddressBook shows an error message <br>
   Use case resumes at step 2
+  
+#### Use case: Rename tag
+
+**Precondition**
+
+All process working fine. 
+
+**Guarantees**
+
+The tag is renamed.
+
+**MSS**
+
+1. User enters command 'rename tag'
+2. AddressBook shows a list of tags ordered with index
+3. User enters the index of the tag that is needed to be renamed, and the new tag name with a space in between
+4. AddressBook displays the message that the tag is renamed successfully <br>
+Use case ends.
+
+**Extensions**
+
+2a. The list is empty
```

Comment:
In use case extensions, the numbering is use to describe scenarios that can happen, preventing the user from going to the next step.
so, this is better off being 1a.